frontend/src/features/admin-form/settings/components/MultiLanguageSection/FormMultiLanguageToggle.tsx

```diff
@@ -0,0 +1,202 @@
+            <Badge variant="solid" colorScheme="success" color="secondary.700">
+              Default
+            </Badge>
+          )}
+        </Flex>
+        {!isDefaultLanguage && (
+          <HStack spacing="0.75rem">
+            <IconButton
+              variant="clear"
+              icon={
+                isLanguageSupported ? (
+                  <GoEye width="44px" />
+                ) : (
+                  <GoEyeClosed width="44px" />
+                )
+              }
+              colorScheme="secondary"
+              aria-label={`Select ${unicodeLocale} as the form's default language`}
+              onClick={() => handleToggleSupportedLanague(unicodeLocale)}
+            />
+            <IconButton
+              variant="clear"
+              icon={<BiEditAlt width="44px" />}
+              colorScheme="secondary"
+              aria-label={`Add ${unicodeLocale} translations`}
+              // TODO: Will add redirection to translation section in next PR
+              onClick={() => console.log('Edit translations')}
+            />
+          </HStack>
+        )}
+      </Flex>
+      {!isLast && <Divider />}
+    </>
+  )
+}
+
+const LanguageTranslationSection = ({
+  defaultLanguage,
+}: LanguageTranslationSectionProps): JSX.Element => {
+  let unicodeLocales = Object.values(Language)
+
+  const idxToRemove = unicodeLocales.indexOf(defaultLanguage)
+  unicodeLocales.splice(idxToRemove, 1)
+
+  unicodeLocales = [defaultLanguage, ...unicodeLocales]
+
+  return (
+    <Box pt={8}>
+      {unicodeLocales.map((unicodeLocale, id, arr) => {
+        return (
+          <LanguageTranslationRow
+            unicodeLocale={unicodeLocale}
+            isDefaultLanguage={unicodeLocale === defaultLanguage}
+            isLast={id === arr.length - 1}
+            key={unicodeLocale}
+          />
+        )
+      })}
+    </Box>
+  )
+}
+
+export const FormMultiLanguageToggle = (): JSX.Element => {
+  const { data: settings, isLoading: isLoadingSettings } =
+    useAdminFormSettings()
+
+  const { mutateFormSupportedLanguages, mutateFormHasMultiLang } =
+    useMutateFormSettings()
+
+  const hasMultiLang = useMemo(
+    () => settings && settings?.hasMultiLang,
+    [settings],
+  )
+
+  const handleToggleMultiLang = useCallback(() => {
+    if (!settings || isLoadingSettings || mutateFormHasMultiLang.isLoading)
+      return
+
+    const nextHasMultiLang = !settings.hasMultiLang
+
+    const currentSupportedLanguages = settings?.supportedLanguages
+
+    // Restore back previously supported languages.
+    let nextSupportedLanguages = currentSupportedLanguages
+
+    // Add all languages as this is the first instance user turns on
+    // toggle
+    if (_.isEmpty(currentSupportedLanguages) && nextHasMultiLang) {
+      nextSupportedLanguages = Object.values(Language)
+    }
+
+    mutateFormSupportedLanguages.mutate({ nextSupportedLanguages })
+
+    return mutateFormHasMultiLang.mutate(nextHasMultiLang)
```

Comment:
not sure but would this cause an inconsistency error if error occurs for this but previous mutation has succeeded?

Reply:
Thanks for considering this scenario. I'll combine the two mutations into a single operation, ensuring that both must succeed; otherwise, the entire mutation will be treated as failed.

Reply:
Thanks for bringing up this scenario. I changed the implementation to only add the array of supported languages in the first instance that the user turns on the toggle for this feature. I also updated the model to have a default false value for the `hasMultiLang` attribute. Hence now if the mutation to update the `hasMultiLang` attribute fail, the data is not in an inconsistent state because the intended implementation was to still keep the list of supported languages even if the `hasMultiLang` value has not been updated. This is to be able to retrieve back the list of supported languages when the user turns the toggle back on.

I hope this clarifies your question.